### PR TITLE
Fixes for GHC 8.8.1

### PR DESCRIPTION
**Pull request by vaibhavsagar**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by cdepillabout** [comment]
@GrahamcOfBorg build haskell.packages.ghc881.foundation haskell.packages.ghc881.socks haskell.packages.ghc881.aeson haskell.packages.ghc881.polyparse haskell.packages.ghc881.cabal-doctest haskell.packages.ghc881.dlist haskell.packages.ghc881.happy haskell.packages.ghc881.attoparsec haskell.packages.ghc881.th-lift haskell.packages.ghc881.th-abstraction haskell.packages.ghc881.unordered-containers haskell.packages.ghc881.haskell-src-exts

File: pkgs/development/haskell-modules/configuration-ghc-8.8.x.nix [hunk 1]
```diff
@@ -82,18 +81,19 @@ self: super: {
...
-  });
   vector-th-unbox = appendPatch super.vector-th-unbox (pkgs.fetchpatch {
     url = "https://raw.githubusercontent.com/hvr/head.hackage/master/patches/vector-th-unbox-0.2.1.6.patch";
     sha256 = "0169yf9ms1g5mmkc5l6hpffzm34zdrqdng4df02nbdmfgba45h19";
   });
-  cabal-doctest = appendPatch  (doJailbreak super.cabal-doctest) (pkgs.fetchpatch {
-    url = "https://raw.githubusercontent.com/hvr/head.hackage/master/patches/cabal-doctest-1.0.6.patch";
-    sha256 = "0735mkxhv557pgnfvdjakkw9r85l5gy28grdwg929m26ghbf9s8j";
-  });
+  cabal-doctest = overrideSrc (doJailbreak super.cabal-doctest) {
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-ghc-8.8.x.nix]: There is a `cabal-doctest_1_0_7` derivation.  Could that be used instead?

**Review by cdepillabout — APPROVED** [review]
LGTM other than the `cabal-doctest_1_0_7` thing.

**Comment by cdepillabout** [comment]
@GrahamcOfBorg build haskell.packages.ghc881.memory haskell.packages.ghc881.cabal-doctest

**Review by basvandijk — APPROVED** [review]

**Comment by vaibhavsagar** [comment]
Thanks!

**Pull request merged**